packages/calcite-components/src/components/dropdown/dropdown.tsx

```diff
@@ -562,30 +561,14 @@ export class Dropdown
 
   private getMaxScrollerHeight(): number {
     const { maxItems, items } = this;
-    let itemsToProcess = 0;
-    let maxScrollerHeight = 0;
-    let groupHeaderHeight: number;
-
-    this.groups.forEach((group) => {
-      if (maxItems > 0 && itemsToProcess < maxItems) {
-        Array.from(group.children).forEach((item: DropdownItem["el"], index) => {
-          if (index === 0) {
-            if (isNaN(groupHeaderHeight)) {
-              groupHeaderHeight = item.offsetTop;
-            }
-
-            maxScrollerHeight += groupHeaderHeight;
-          }
-
-          if (itemsToProcess < maxItems) {
-            maxScrollerHeight += item.offsetHeight;
-            itemsToProcess += 1;
-          }
-        });
-      }
-    });
 
-    return items.length >= maxItems ? maxScrollerHeight : 0;
+    return items.length >= maxItems ? this.getYDistance(this.scrollerEl, items[maxItems - 1]) : 0;
+  }
+
+  private getYDistance(parent: HTMLElement, child: HTMLElement): number {
+    const parentRect = parent.getBoundingClientRect();
+    const childRect = child.getBoundingClientRect();
+    return childRect.bottom - parentRect.top;
```

Comment:
I'm new to the scroller logic and this function took me a second to wrap my head around. A couple questions:

- Is the scroller the parent element of the dropdown items? That's what it sounds like from the variable names.
- Would the top of the first item be equivalent to the top of the scroller? If so, using the first item instead would probably make it easier to read.